### DISCUSSION TITLE
Unity

### DISCUSSION THREAD

**Discussion by nguyenkduywork** [Web3Auth/Web3Auth#385] [Help]
Hi, I would like to know a few things:

1. How can I see the public address in the game (just show it in the console would be great for now)
2. I would like to call some solidity functions inside the game too. I used to use Chainsafe, but the thing is they limit us to Metamask only. Are there any other ways that I can call solidity functions after login in by using Web3Auth..?

An example would be: I use **web3auth to log in**, and I get my **public address**. I want to test with a Mint function which will mint 100 tokens into that public address. (The mint function is made in the most simple way possible, anyone can use the function, for now. It has two arguments: **public address that I get from logging in with Web3Auth** and amount to mint). 

----> How can I call that function?

Thank you in advance!

**Comment by hugtor** [comment]
Hi did you post a qn in the other thread as well? I can't find it now, could you repost here and I can ans together?

**Reply by nguyenkduywork** [reply]
Hey! I deleted that question :)

**Reply by nguyenkduywork** [reply]
...whenever you are ready! :)

**Reply by hugtor** [reply]
1. how you get the public address depends on the blockchain. are you using mainly eth?
2. hmm at the moment that isnt supported yet. one possible workaround is if you're building for android/ios you can import the java/swift/obj-c code into unity and use that.

**Reply by nguyenkduywork** [reply]
1. Hey, yes I mainly use ETH for now.

2. Oh okay, I don't have much experience doing that (importing other coding languages into unity), also I think it would not be easy to maintain the code should the SDK get updated, but thanks for your help Hugtor :)
